Core/Src/main_task.c

```diff
@@ -448,20 +540,31 @@ static void run_upshift_sm(void)
 		set_upshift_solenoid(SOLENOID_ON);
 		safe_spark_cut(true);
 
+#ifdef NO_GEAR_POT
 		// the shifter has moved above the threshold of exiting the gear. Spark
 		// must be cut again to reach the correct RPM for the next gear. If
 		// enough time has passed return spark anyway
 		if (get_shift_pot_pos() > UPSHIFT_EXIT_POS_MM ||
 				HAL_GetTick() - begin_exit_gear_spark_return_tick > UPSHIFT_EXIT_SPARK_RETURN_MS)
 		{
-#ifdef SHIFTDEBUG
+#else
+		if (tcm_data.current_gear > initial_gear)
```

Comment:
Even with the shift position sensor you will need to have a timeout on this

Reply:
? The timeout is still below this